test/test_mahalanobis_mixin.py

```diff
@@ -25,7 +26,7 @@ def test_score_pairs_pairwise(estimator, build_dataset):
   X = X[:n_samples]
   model = clone(estimator)
   set_random_state(model)
-  model.fit(input_data, labels)
+  model.fit(*make_args_inc_quadruplets(estimator, input_data, labels))
```

Comment:
this quite ugly ;-) if such a function is really handy, maybe try to at least find a shorter name

Reply:
I think such a function is needed because we don't have y for quadruplets learners
There might be a way to change this like making our `build_dataset` function return some "dataset" object that would be `(a, b)` (like `(X, y), (pairs, y_pairs)`...) for most estimators, but `(quadruplets,)` for quadruplets. Then we could pass it like this: `fit(*dataset)`. But then it would be problematic say if we want to split the dataset: `train_test_split(dataset)` would return either 4 or 2 objects, and it'll become hard to separate them... So I think we should deal with forming the appropriate object just before using them with `fit`, to have more freedom to control the objects before...
I'm not 100% sure about this, but I think it's OK to have this function for now, and anyway it shouldn't be too difficult to change later if we find a better solution

And I agree, the name is really bad :p I'll change it, what about `remove_y_quadruplets` ?

Reply:
Yes, renaming is gine